docs/gitbook/tutorials/skipper-progressive-delivery.md

```diff
@@ -0,0 +1,382 @@
+ Pre-rollout check acceptance-test passed
+ Advance podinfo.test canary weight 5
+ Advance podinfo.test canary weight 10
+ Advance podinfo.test canary weight 15
+ Advance podinfo.test canary weight 20
+ Advance podinfo.test canary weight 25
+ Advance podinfo.test canary weight 30
+ Advance podinfo.test canary weight 35
+ Advance podinfo.test canary weight 40
+ Advance podinfo.test canary weight 45
+ Advance podinfo.test canary weight 50
+ Copying podinfo.test template spec to podinfo-primary.test
+ Waiting for podinfo-primary.test rollout to finish: 1 of 2 updated replicas are available
+ Routing all traffic to primary
+ Promotion completed! Scaling down podinfo.test
+```
+
+**Note** that if you apply new changes to the deployment during the canary analysis, Flagger will restart the analysis.
+
+You can monitor all canaries with:
+
+```bash
+watch kubectl get canaries --all-namespaces
+
+NAMESPACE   NAME        STATUS        WEIGHT   LASTTRANSITIONTIME
+test        podinfo-2   Progressing   30       2020-08-14T12:32:12Z
+test        podinfo     Succeeded     0        2020-08-14T11:23:88Z
+```
+
+## Automated rollback
+
+During the canary analysis you can generate HTTP 500 errors to test if Flagger pauses and rolls back the faulted version.
+
+Trigger another canary deployment:
+
+```bash
+kubectl -n test set image deployment/podinfo \
+podinfod=stefanprodan/podinfo:4.0.6
+```
+
+Exec into the load tester pod with:
+
+```bash
+kubectl -n test exec -it deploy/flagger-loadtester bash
+```
+
+Generate HTTP 500 errors:
+
+```bash
+hey -z 1m -c 5 -q 5 http://app.example.com/status/500
+```
+
+Generate latency:
+
+```bash
+watch -n 1 curl http://app.example.com/delay/1
+```
+
+When the number of failed checks reaches the canary analysis threshold, the traffic is routed back to the primary,
+the canary is scaled to zero and the rollout is marked as failed.
+
+```text
+kubectl -n flagger-system logs deploy/flagger -f | jq .msg
+
+"New revision detected! Scaling up podinfo.test"
+"GetRoutes primaryWeight: 100, canaryWeight: 0"
+"canary deployment podinfo.test not ready: waiting for rollout to finish: 0 of 1 updated replicas are available"
+"GetRoutes primaryWeight: 100, canaryWeight: 0"
+"Starting canary analysis for podinfo.test"
+"Pre-rollout check acceptance-test passed"
+"primaryWeight: 95, canaryWeight: 5"
+"Advance podinfo.test canary weight 5"
+"GetRoutes primaryWeight: 95, canaryWeight: 5"
+"primaryWeight: 90, canaryWeight: 10"
+"Advance podinfo.test canary weight 10"
+"GetRoutes primaryWeight: 90, canaryWeight: 10"
+"primaryWeight: 85, canaryWeight: 15"
+"Advance podinfo.test canary weight 15"
+"GetRoutes primaryWeight: 85, canaryWeight: 15"
+"primaryWeight: 80, canaryWeight: 20"
+"Advance podinfo.test canary weight 20"
+"GetRoutes primaryWeight: 80, canaryWeight: 20"
+"Halt podinfo.test advancement success rate 53.42% < 99%"
+"GetRoutes primaryWeight: 80, canaryWeight: 20"
```

Comment:
The GetRoutes logs are debug only right? Can you please remove them from the example.